benchmark/benchncnn.cpp

```diff
@@ -380,82 +419,52 @@ int main(int argc, char** argv)
+        run_benchmark_if_supported("squeezenet", ncnn::Mat(227, 227, 3), opt);
+        run_benchmark_if_supported("squeezenet_int8", ncnn::Mat(227, 227, 3), opt);
+        run_benchmark_if_supported("mobilenet", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("mobilenet_int8", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("mobilenet_v2", ncnn::Mat(224, 224, 3), opt);
+        // run_benchmark_if_supported("mobilenet_v2_int8", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("mobilenet_v3", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("shufflenet", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("shufflenet_v2", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("mnasnet", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("proxylessnasnet", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("efficientnet_b0", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("efficientnetv2_b0", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("regnety_400m", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("blazeface", ncnn::Mat(128, 128, 3), opt);
+        run_benchmark_if_supported("googlenet", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("googlenet_int8", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("resnet18", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("resnet18_int8", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("alexnet", ncnn::Mat(227, 227, 3), opt);
+        run_benchmark_if_supported("vgg16", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("vgg16_int8", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("resnet50", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("resnet50_int8", ncnn::Mat(224, 224, 3), opt);
+        run_benchmark_if_supported("squeezenet_ssd", ncnn::Mat(300, 300, 3), opt);
+        run_benchmark_if_supported("squeezenet_ssd_int8", ncnn::Mat(300, 300, 3), opt);
+        run_benchmark_if_supported("mobilenet_ssd", ncnn::Mat(300, 300, 3), opt);
+        run_benchmark_if_supported("mobilenet_ssd_int8", ncnn::Mat(300, 300, 3), opt);
+        run_benchmark_if_supported("mobilenet_yolo", ncnn::Mat(416, 416, 3), opt);
+        run_benchmark_if_supported("mobilenetv2_yolov3", ncnn::Mat(352, 352, 3), opt);
+        run_benchmark_if_supported("yolov4-tiny", ncnn::Mat(416, 416, 3), opt);
+        run_benchmark_if_supported("nanodet_m", ncnn::Mat(320, 320, 3), opt);
+        run_benchmark_if_supported("yolo-fastest-1.1", ncnn::Mat(320, 320, 3), opt);
+        run_benchmark_if_supported("yolo-fastestv2", ncnn::Mat(352, 352, 3), opt);
+        run_benchmark_if_supported("vision_transformer", ncnn::Mat(384, 384, 3), opt);
+        run_benchmark_if_supported("FastestDet", ncnn::Mat(352, 352, 3), opt);
```

Comment:
Add the logic of `run_benchmark_if_supported` directly to the benchmark() function so that we don’t need to modify the call here.run_benchmark_if_supported